services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainController.java

```diff
@@ -0,0 +1,198 @@
+import org.apache.tuweni.bytes.Bytes;
+import tech.pegasys.artemis.datastructures.Constants;
+import tech.pegasys.artemis.metrics.ArtemisMetricCategory;
+import tech.pegasys.artemis.metrics.SettableGauge;
+import tech.pegasys.artemis.networking.p2p.JvmLibP2PNetwork;
+import tech.pegasys.artemis.networking.p2p.MockP2PNetwork;
+import tech.pegasys.artemis.networking.p2p.api.P2PNetwork;
+import tech.pegasys.artemis.networking.p2p.jvmlibp2p.JvmLibp2pConfig;
+import tech.pegasys.artemis.statetransition.StateProcessor;
+import tech.pegasys.artemis.statetransition.events.GenesisEvent;
+import tech.pegasys.artemis.statetransition.events.ValidatorAssignmentEvent;
+import tech.pegasys.artemis.storage.ChainStorage;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+import tech.pegasys.artemis.storage.Store;
+import tech.pegasys.artemis.storage.events.DBStoreValidEvent;
+import tech.pegasys.artemis.storage.events.NodeStartEvent;
+import tech.pegasys.artemis.storage.events.SlotEvent;
+import tech.pegasys.artemis.util.alogger.ALogger;
+import tech.pegasys.artemis.util.config.ArtemisConfiguration;
+import tech.pegasys.artemis.util.time.Timer;
+import tech.pegasys.artemis.util.time.TimerFactory;
+import tech.pegasys.artemis.validator.coordinator.ValidatorCoordinator;
+import tech.pegasys.pantheon.metrics.MetricsSystem;
+
+public class BeaconChainController {
+  private static final ALogger STDOUT = new ALogger("stdout");
+  private final ArtemisConfiguration config;
+  private EventBus eventBus;
+  private Vertx vertx;
+  private Timer timer;
+  private ChainStorageClient chainStorageClient;
+  private P2PNetwork p2pNetwork;
+  private final MetricsSystem metricsSystem;
+  private SettableGauge currentSlotGauge;
+  private SettableGauge currentEpochGauge;
+  private ValidatorCoordinator validatorCoordinator;
+  private StateProcessor stateProcessor;
+
+  private UnsignedLong nodeSlot = UnsignedLong.ZERO;
+
+  public BeaconChainController(
+      EventBus eventBus, Vertx vertx, MetricsSystem metricsSystem, ArtemisConfiguration config) {
+    this.eventBus = eventBus;
+    this.vertx = vertx;
+    this.config = config;
+    this.metricsSystem = metricsSystem;
+    this.eventBus.register(this);
+  }
+
+  @SuppressWarnings("rawtypes")
+  public void initTimer() {
+    int timerPeriodInMiliseconds = (int) ((1.0 / Constants.TIME_TICKER_REFRESH_RATE) * 1000);
+    try {
+      this.timer =
+          new TimerFactory()
+              .create(
+                  config.getTimer(),
+                  new Object[] {this.eventBus, 0, timerPeriodInMiliseconds},
+                  new Class[] {EventBus.class, Integer.class, Integer.class});
+    } catch (IllegalArgumentException e) {
+      System.exit(1);
+    }
+  }
+
+  public void initStorage() {
+    this.chainStorageClient = ChainStorage.Create(ChainStorageClient.class, eventBus);
+    this.chainStorageClient.setGenesisTime(UnsignedLong.valueOf(config.getInteropGenesisTime()));
```

Comment:
an if statement here to see if we're actually in interop mode might be useful. also, how are you going to set the time if we're not in interop mode?

Reply:
yes, let's talk this through

Reply:
Also, I need the config.toml genesis time now for interop mode and when running a 4 node demo.   Thinking about renaming it

Reply:
This is the one thing i didn't fully resolve.  I think we need to rethink how our config options are grouped so it is clear what options are for 'dev' and what options are for 'prod'